_posts/2022-12-09-aws-sigv4-support-for-clients.md

```diff
@@ -0,0 +1,176 @@
+
+```ruby
+require 'opensearch-aws-sigv4'
+require 'aws-sigv4'
+signer = Aws::Sigv4::Signer.new(service: 'es',
+                                region: 'us-west-2',
+                                access_key_id: 'key_id',
+                                secret_access_key: 'secret')
+client = OpenSearch::Aws::Sigv4Client.new({ log: true }, signer)
+client.cluster.health
+client.transport.reload_connections!
+client.search q: 'test'
+```
+
+### Creating a client connection in .NET
+
+All required request signing is handled by the `AwsSigV4HttpConnection` implementation, by default it will use the .NET AWS SDK's default credentials provider 
+to acquire credentials from the environment. However, you may opt to pass in your own credentials provider; e.g. to assume a role. Refer to the [OpenSearch.Net User Guide](https://github.com/opensearch-project/opensearch-net/blob/main/USER_GUIDE.md#opensearchnetauthawssigv4) for complete getting started instructions.
+
+```c#
+using OpenSearch.Client;
+using OpenSearch.Net.Auth.AwsSigV4;
+
+var endpoint = new Uri("https://search-xxx.region.es.amazonaws.com");
+var connection = new AwsSigV4HttpConnection();
+var config = new ConnectionSettings(endpoint, connection);
+var client = new OpenSearchClient(config);
+```
+
+### Creating a client connection in Rust
+
+Request signing is configured using the [`Credentials::AwsSigV4`](https://docs.rs/opensearch/latest/opensearch/auth/enum.Credentials.html#variant.AwsSigV4) enum variant or its helper conversion from an AWS SDK config. See [aws-config](https://docs.rs/aws-config/latest/aws_config/) for other AWS credentials provider implementations; e.g. assume role.
+
+```rust
+use opensearch::{
+    cat::CatIndicesParts,
+    http::transport::{SingleNodeConnectionPool, TransportBuilder},
+    OpenSearch,
+};
+use url::Url;
+
+let creds = aws_config::load_from_env().await;
+
+let host = "https://search-xxx.region.es.amazonaws.com";
+let transport = TransportBuilder::new(SingleNodeConnectionPool::new(Url::parse(host).unwrap()))
+    .auth(creds.try_into()?)
+    // .auth(Credentials::AwsSigV4(creds.credentials().unwrap().clone(), creds.region().unwrap().clone()))
+    .build()?;
+let client = OpenSearch::new(transport);
+```
+
+## Conclusion
+
+You can now sign their requests using the client APIs natively instead of using workarounds. We’re continuing to work on improving the capabilities of SigV4 in clients with scenarios like async connections, compressed requests, and connection pooling support. We’re happy to take pull requests and feedback in the form of issues on [github](https://github.com/opensearch-project/opensearch-py/issues).
```

Comment:
```suggestion
You can now sign your requests using the client APIs natively instead of using workarounds. We’re continuing to work on improving the capabilities of SigV4 in clients with scenarios like async connections, compressed requests, and connection pooling support. We’re happy to take pull requests and feedback in the form of issues on [github](https://github.com/opensearch-project/opensearch-py/issues).
```